kamon-scala-future/src/main/scala/kamon/instrumentation/futures/scala/ScalaFutureInstrumentation.scala

```diff
@@ -78,9 +155,37 @@ object ScalaFutureInstrumentation {
     *            bytecode instrumentation. If instrumentation is disabled you risk leaving dirty threads that can cause
     *            incorrect Context propagation behavior.
     */
-  def traceAsync[T, S](operationName: String)(body: T => S)(implicit settings: Settings): T => S = { t: T =>
-    val span = startedSpan(operationName, settings)
-    Kamon.store(Kamon.currentContext().withKey(Span.Key, span))
+  def traceFun[T, S](operationName: String, tags: TagSet = TagSet.Empty, metricTags: TagSet = TagSet.Empty)(body: T => S)
```

Comment:
What about `traceFunc`? Would be a little bit more clear that it traces functions.